app/pages/_app.tsx

```diff
@@ -68,13 +68,13 @@ export default class App extends NextApp<AppProps> {
 
     return (
       <Sentry.ErrorBoundary fallback={ErrorFallback}>
-        {Component.isAccessProtected && (
-          <SessionTimeoutHandler modalDisplaySecondsBeforeLogout={120} />
-        )}
         <PageRedirectHandler
           environment={environment}
           pageComponent={Component}
         >
+          {Component.isAccessProtected && (
+            <SessionTimeoutHandler modalDisplaySecondsBeforeLogout={120} />
+          )}
```

Comment:
This might reverse the problem and always show the "You  must be logged in" message instead of "your session timed out" when appropriate

Reply:
Will be addressed in #2143